docs/painless/painless-execute-script.asciidoc

```diff
@@ -0,0 +1,53 @@
+[[painless-execute-api]]
+=== Painless execute API
+
+The Painless execute API allows to execute an arbitrary script and introspect the result.
+
+[[painless-execute-api-parameters]]
+.Parameters
+[options="header"]
+|======
+| Name             | Required  | Default                | Description
+| `script`         | yes       | -                      | The script to execute
+| `context`        | no        | `execute_api_script`   | The context the script should be executed in.
+|======
+
+==== Contexts
+
+Contexts control how scripts get executed, what variables are available at runtime and what the return type is.
+
+===== Execute api script context
+
+The `execute_api_script` context executes scripts as is and do not add any special variables.
+The only variable that is available is `params`, which can be used to access script parameters.
+The result of the script is always converted to a string.
+If no context is specified then this context is used by default.
+
+==== Example
+
+Request:
+
+[source,js]
+----------------------------------------------------------------
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": "params.count / params.total",
+    "params": {
+      "count": 100.0,
+      "total": 1000.0
+    }
+  }
+}
+----------------------------------------------------------------
+// CONSOLE
+
+Response:
+
+[source,js]
+--------------------------------------------------
+{
+  "result": "0.1"
```

Comment:
Should there be a way to possibly return specified parameters as well so they the user can see side-effects?  Maybe a follow up.

Reply:
Not sure, whether returning the specified variables makes sense.

Reply:
Yeah it probably doesn't make sense here.  I guess I was thinking along the lines of how things get passed around in related to aggs, but that can be simulated anyway.